src/main/java/core/basesyntax/service/strategy/ReturnFruitTransactionHandler.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.service.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+
+public class ReturnFruitTransactionHandler implements FruitTransactionHandler {
+    private Storage storage;
+
+    public ReturnFruitTransactionHandler(Storage storage) {
+        this.storage = storage;
+    }
+
+    @Override
+    public void handler(FruitTransaction fruitTransaction) {
+        storage.add(fruitTransaction.getFruit(), fruitTransaction.getQuantity());

```

The handler delegates directly to `storage.add(...)` without validating the transaction fields. Even if the converter is expected to validate input, handlers should fail fast with clear messages when receiving invalid data. Validate `fruitTransaction.getFruit()` (not null/blank) and `fruitTransaction.getQuantity()` (>= 0) and throw explicit exceptions when invalid.